src/librbd/journal/Types.cc

```diff
@@ -384,11 +385,15 @@ void EventEntry::decode(bufferlist::iterator& it) {
   }
 
   boost::apply_visitor(DecodeVisitor(struct_v, it), event);
+  if (struct_v >= 4) {
+    ::decode(timestamp, it);
```

Comment:
@dillaman Thinking about this a little more, not sure it will always work. If a new event type is added (or an existent one is extended) I expect it will fail.

Actually, I would like timestamp be encoded at the beginning of the EventEntry, before Event, but I don't see how to preserve compatibility, so an older client still could read a journal created by a new client. A way I see is to 
make timestamp a member of Event, i.e. adding to every event structure (or use inheritance). I am not like this much though.

Do you have any suggestions? Do we want older clients to be able to decode newer journals?